src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -179,11 +179,9 @@ private void AddErrorLabelIfRequired(Exception exception, SemanticVersion server
                     mongoException.AddErrorLabel("TransientTransactionError");
                 }
 
-                if ((exception is MongoConnectionException) || // network error
-                    (Feature.RetryableWrites.IsSupported(serverVersion) &&
-                    !Feature.ServerReturnsRetryableWriteErrorLabel.IsSupported(serverVersion)))
```

Comment:
Can you elaborate on this completely different test?

My best guess is that before we would try to determine if the operation *should* have been retryable, but now we are looking inside the actual command to see if it *actually* is retryable?

Reply:
In the previous commit: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/123/commits/7694a47c327e1107f4aa836e0f076498b81f83a9#diff-91a7ef566b7da37c92d9773b00943b16R182 we had this logic:

    if (_command.Contains("txnNumber"))
    {
      RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired(mongoException, serverVersion);
    }

where the validation on txtNumber meant that we want to add the retryable write label only for operations with configured `retryWrites=true`. But we also need to add this label for commitTransaction and abortTransaction which are also retryable regardless the retryWrites option. Spec:

>Drivers MUST retry the commitTransaction and abortTransaction commands even when retryWrites has been disabled on the MongoClient

https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#interaction-with-retryable-writes

Reply:
Instead of comparing to the last commit can you compare the changes to master?

Reply:
The whole logic from the master was moved inside `RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired`, so no real changes in this part. 
In the previous commit, in addition to the conditions that are already in master, I've added the txtNumber condition to add the write error label only for operations with configured `retryWrites=true`. In the current version, additionally to this condition, I've added two more: 
* on commitTransaction
* on abortTransaction
because these operations are retryable regardless of `retryWrites=true`.

Reply:
as I wrote here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/123#issuecomment-612541406, It can be not the best approach, but it's the simplest

Reply:
I was hoping you would summarize the change in one place here.

I will follow all your links and see if I can figure out what you are saying.

Reply:
I think I summarized it :) , the whole point that now we have 3 more additional conditions for the logic which is in master now.
These 3 conditions are:
* check that retryWrites=true is configured
* the operation is commitTransaction
* the operation is abortTransaction

Reply:
I don't think your statement is true. 

We don't have 3 ADDITIONAL conditions. What we have is 3 NEW conditions that completely REPLACE the previous conditions.

This is what I said before: My best guess is that before we would try to determine if the operation *should* have been retryable, but now we are looking inside the actual command to see if it *actually* is retryable?

Is what I said true? If not, can you summarize why you completely REPLACED the logic with new logic?

Reply:
>We don't have 3 ADDITIONAL conditions. What we have is 3 NEW conditions that completely REPLACE the previous conditions.

no, it's not true. As I wrote here: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/123#discussion_r408103022 the previous comparing logic based on server versions and error codes was moved inside `RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired`. See: https://github.com/DmitryLukyanov/mongo-csharp-driver/pull/123/files#diff-f3cf36bc2821907b3003ced21fae1593R166-R181
Now, the selected lines contain only new conditions based on the operation name and operation options.

Reply:
Oh that's tricky... you didn't move them to `RetryabilityHelper.IsCommandRetryable`

You moved them to `RetryabilityHelper.AddRetryableWriteErrorLabelIfRequired`

No wonder I didn't realize what you had done.

Reply:
:)

Reply:
I'm going to resolve this comment. Thanks for the explanations, even if they didn't make sense to me at first.